libsolidity/formal/SMTEncoder.cpp

```diff
@@ -923,23 +923,22 @@ void SMTEncoder::endVisit(IndexAccess const& _indexAccess)
 
 	if (_indexAccess.annotation().type->category() == Type::Category::TypeType)
 		return;
+	if (_indexAccess.baseExpression().annotation().type->category() == Type::Category::FixedBytes)
+	{
+		m_errorReporter.warning(
+			7989_error,
+			_indexAccess.location(),
+			"Assertion checker does not yet support index accessing fixed bytes."
+		);
+		return;
+	}
```

Comment:
Shouldn't there be a similar check for `Type::Category::Array`?

Reply:
Accessing the index of an array is fine, so I'm not sure I understand why a check for Array

Reply:
I think I am a bit confused. Isn't the base type of this variable a bytes array (and hence of type `Type::Category::Array`)? 

I got this doubt because the docu (https://solidity.readthedocs.io/en/v0.6.12/types.html#bytes-and-strings-as-arrays) states that `bytes` and `string` are array types

Reply:
Yes, and that's why accessing `x[8]` works in the SMTChecker. But the returned type is FixedBytes array, that's why accessing an index of the returned expression failed before

Reply:
Ah I see. So IIUC `x[8]` resolves to an expression of `byte` (aka `bytes1`) and the second index is accessing an offset of this type.

If this is the case, shouldn't the compiler complain about second index accesses of `bytes` type because afaiu `bytes` is an array of single bytes, and single bytes cannot have offsets i.e., there are not arrays in themselves?

Reply:
Yes, exactly!
Yea we had a short discussion about that on Gitter.
Accessing bytes gives you a `byte` which is an alias to `bytes1`. Since `bytes1` belongs to the `bytesNN` family, it can be accessed via index, in this case only 0.

Reply:
@chriseth argued `I would still keep it for symmetry reasons, especially thinking of templates...`